src/test/tor/minimal/run.sh

```diff
@@ -0,0 +1,4 @@
+# Ensure we have the correct permissions on the hs dir, or Tor won't run
+chmod 700 shadow.data.template/hosts/hiddenserver/hs
```

Comment:
The "chmod" step is done as part of `./setup build` when it copies the files from `src/` to `build/src`. The build step also fixes the `AuthDirTestReachability` option depending on the tor version. When running shadow below, would it be better to use the template directory in `build/src` instead? Or maybe this script should run `(~/.local/bin/tor --list-torrc-options | grep -q 'AuthDirTestReachability') || find conf -type f -print0 | xargs -0 sed -i '/AuthDirTestReachability/s/^/#/g` itself?

Reply:
> The "chmod" step is done as part of ./setup build when it copies the files from src/ to build/src. The build step also fixes the AuthDirTestReachability option depending on the tor version.

Oh...
- Somehow I got myself into a situation where the test was failing for me when I ran the run script without the chmod. Probably because I was moving around branches or something and didn't rebuild before running the test.
- I think it will be pretty easy for users to get into that situation, so doing whatever `chmod`s or tor version checks at runtime I think is the best way to maximize chance of success.
- It is not obvious to the user that they should need to make sure they have pinned a Tor version when they compile shadow. That is non-standard and fragile I think. What happens if they want to update Tor later but then run the shadow test again without rebuilding?

I guess some of these issues are arising because I am trying to hijack our CI test and turn it into an example that users can run to quickly get some results from a small Tor network.

Reply:
Okay, I think having the run script makes sense. Maybe it shouldn't run the grep/find/sed command above then, so that it doesn't modify the torrc files and break the test's CMake file.

Also my PR to make tor not require special permissions on these files when running with `TestingTorNetwork` was never looked at, so I don't think this behaviour will be changing.

Reply:
> Maybe it shouldn't run the grep/find/sed command above then, so that it doesn't modify the torrc files and break the test's CMake file.

Not sure I follow this part. I think you might mean we may want to modify the `CMakeLists.txt` file that does the find/replace of the `AuthDirTestReachability` option. If so, I think it would be reasonable to remove it and then just document that our test case requires that you run Tor v0.x.x.x or later. I think we already have a similar requirement since our torrc files use `ORPort 9111 IPv4Only`, so I would be fine with this. Probably should do it in a separate PR though.

> Also my PR to make tor not require special permissions on these files when running with TestingTorNetwork was never looked at, so I don't think this behaviour will be changing.

That's too bad :( I added a comment in support:
https://gitlab.torproject.org/tpo/core/tor/-/issues/40338

Reply:
> Not sure I follow this part. I think you might mean we may want to modify the CMakeLists.txt file that does the find/replace of the AuthDirTestReachability option. If so, I think it would be reasonable to remove it and then just document that our test case requires that you run Tor v0.x.x.x or later. I think we already have a similar requirement since our torrc files use ORPort 9111 IPv4Only, so I would be fine with this. Probably should do it in a separate PR though.

Our CI tests run all supported versions of tor (except the newest in #1453), which require us to handle the different `AuthDirTestReachability` behaviours. So I think we have to do this find/replace in the CMakeLists.txt file.

Maybe we should just copy these files to an examples directory and have the example and test separate?

Reply:
I think we should leave it as it is in this PR. I don't think the Tor test is run by default when running `./setup test`, so most people won't run it anyway and there won't be any conflicts when they run the Tor example. If it does end up causing problems for people, we could change it up at that point.

Reply:
Sound OK?

Reply:
Sounds good